nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/PutS3Object.java

```diff
@@ -517,331 +526,328 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         /*
          * Then
          */
-        try {
-            final FlowFile flowFileCopy = flowFile;
-            session.read(flowFile, new InputStreamCallback() {
-                @Override
-                public void process(final InputStream in) throws IOException {
-                    final ObjectMetadata objectMetadata = new ObjectMetadata();
-                    objectMetadata.setContentLength(ff.getSize());
-
-                    final String contentType = context.getProperty(CONTENT_TYPE)
-                            .evaluateAttributeExpressions(ff).getValue();
-                    if (contentType != null) {
-                        objectMetadata.setContentType(contentType);
-                        attributes.put(S3_CONTENT_TYPE, contentType);
-                    }
+        final FlowFile flowFileCopy = session.clone(flowFile);
```

Comment:
I don't think we need to clone the FlowFile. What is the reason for that?

Reply:
Cloning the flowfile was only used as a workaround for an issue which occurred during unit test execution. 
If the inputstream of the flowfile was open, the `session.putAllAttributes(flowFile, attributes)` call threw exception, because the inputstream was already in use. I agree that the cloning is not needed, after some examination, I managed to eliminate the need for that. 
See  73f49bbc